src/Cake.Prca/Issues/CodeAnalysisIssue.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Cake.Prca.Issues
+{
+    using System;
+    using Core.IO;
+
+    /// <summary>
+    /// Base class for a code analysis issue.
+    /// </summary>
+    public class CodeAnalysisIssue : ICodeAnalysisIssue
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CodeAnalysisIssue"/> class.
+        /// </summary>
+        /// <param name="filePath">The path to the file affacted by the issue.
+        /// The path needs to be relative to the repository root.</param>
+        /// <param name="line">The line in the file where the issues has occurred.</param>
+        /// <param name="message">The message of the code analysis issue.</param>
+        /// <param name="priority">The priority of the message used to filter out issues if there are more issues than
+        /// should be posted.</param>
+        /// <param name="rule">The rule of the code analysis issue.</param>
+        public CodeAnalysisIssue(string filePath, int line, string message, int priority, string rule)
+        {
+            filePath.NotNullOrWhiteSpace(nameof(filePath));
+            line.NotNegativeOrZero(nameof(line));
```

Comment:
I think our implementation should consider all issues which could be mapped to a file. We could make line nullable, if no line is defined this means the warning applies to the whole file. The pull request system can then decide where it could put the comment, e.g. TFS could add a comment on the file level. Other PR-systems could abandon them.